### PR TITLE
8 - blockSelection

### LINKED ISSUE #8: blockSelection.sh
Selección de oración y/o párrafo en base a un número de entrada. Es decir, se puede establecer como entrada “O” o “P” para indicar oración o párrafo y luego un número (un párrafo se distingue de otro con un punto y aparte, las oraciones vía un punto seguido).

### PR DESCRIPTION
**Pull request by fedeantuna**

### CODE REVIEW

File: blockSelection.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+### the selection. Those elements will be placed on an array and with
+### a valid input on the second prompt it will display the value on
+### that position.
+
+echo "Select by sentence (O) or paragraph (P)? "
+read SELECTION
+
+SELECTION=$(echo $SELECTION | tr '[:lower:]' '[:upper:]')
+
+# [[ $SELECTION -ne "O" || $SELECTION -ne "P" ]] && exit 1
```
Comment (salvanya) [inline blockSelection.sh]:
```suggestion
[[ $SELECTION -ne "O" || $SELECTION -ne "P" ]] && exit 1
```

File: blockSelection.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+TEXT=$(cat $1)
+ELEMENTS=()
+case $SELECTION in
+    [Oo])
+        DELIMITER='. '
+        TEXT_COPY=$TEXT$DELIMITER
+        while [[ $TEXT_COPY ]]; do
```
Comment (Flor243) [inline blockSelection.sh]:
```suggestion
        while [[ $TEXT_COPY ]] 
        do
```

File: blockSelection.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            ELEMENTS+=("${TEXT_COPY%%"$DELIMITER"*}");
+            TEXT_COPY=${TEXT_COPY#*"$DELIMITER"};
+        done;
```
Comment (Flor243) [inline blockSelection.sh]:
```suggestion
        done
```

File: blockSelection.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+# [[ $SELECTION -ne "O" || $SELECTION -ne "P" ]] && exit 1
+TEXT=$(cat $1)
+ELEMENTS=()
+case $SELECTION in
+    [Oo])
+        DELIMITER='. '
+        TEXT_COPY=$TEXT$DELIMITER
+        while [[ $TEXT_COPY ]]; do
+            ELEMENTS+=("${TEXT_COPY%%"$DELIMITER"*}");
```
Comment (Flor243) [inline blockSelection.sh]:
```suggestion
            ELEMENTS+=("${TEXT_COPY%%"$DELIMITER"*}")
```

Comment (Flor243) [inline blockSelection.sh]:
```suggestion
            ELEMENTS+=("${TEXT_COPY%%"$DELIMITER"*}")
```

File: blockSelection.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            TEXT_COPY=${TEXT_COPY#*"$DELIMITER"};
```
Comment (Flor243) [inline blockSelection.sh]:
```suggestion
            TEXT_COPY=${TEXT_COPY#*"$DELIMITER"}
```

File: blockSelection.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+### a valid input on the second prompt it will display the value on
+### that position.
+
+echo "Select by sentence (O) or paragraph (P)? "
+read SELECTION
+
+SELECTION=$(echo $SELECTION | tr '[:lower:]' '[:upper:]')
+
+[[ $SELECTION -ne "O" || $SELECTION -ne "P" ]] && exit 1
+TEXT=$(cat $1)
```
Comment (Flor243) [inline blockSelection.sh]:
```suggestion
TEXT=$(cat $1 | sed '/^[[:space:]]*$/d')
```
With this change you can avoid reading blank paragraphs :)

**Review by Flor243 — APPROVED** [review]

**Review by salvanya — APPROVED** [review]

**Pull request merged**